server/raft_transport.go

```diff
@@ -101,43 +115,101 @@ func (t *rpcTransport) Stop(id proto.RaftNodeID) {
 	delete(t.servers, id)
 }
 
-// Send a message to the recipient specified in the request.
-func (t *rpcTransport) Send(req *multiraft.RaftMessageRequest) error {
-	// Convert internal to proto formats.
-	protoReq := &proto.RaftMessageRequest{GroupID: req.GroupID}
-	var err error
-	if protoReq.Msg, err = req.Message.Marshal(); err != nil {
-		return err
+func (t *rpcTransport) processQueue(nodeID proto.RaftNodeID) {
+	t.mu.Lock()
+	ch, ok := t.queues[nodeID]
+	t.mu.Unlock()
+	if !ok {
+		return
 	}
-
-	nodeID, _ := proto.DecodeRaftNodeID(proto.RaftNodeID(req.Message.To))
-	addr, err := t.gossip.GetNodeIDAddress(nodeID)
-	if err != nil {
-		return err
+	// Clean-up when the loop below shuts down.
+	defer func() {
+		t.mu.Lock()
+		delete(t.queues, nodeID)
+		t.mu.Unlock()
+	}()
+	var req *multiraft.RaftMessageRequest
+	protoReq := &proto.RaftMessageRequest{}
+	for {
+		select {
+		case <-t.rpcContext.Stopper.ShouldStop():
+			return
+		case <-time.After(raftIdleTimeout):
```

Comment:
This has a race: if we read from the time.After channel, then `ch` is empty at this moment, but `Send` may write something to `ch` before the deferred function fires and discards the channel. Races that simply drop messages aren't crucial since raft will re-send things as necessary, but it would be better to avoid them.